pkg/proxy/iptables/proxier_test.go

```diff
@@ -2495,4 +2556,204 @@ func Test_updateEndpointsMap(t *testing.T) {
 	}
 }
 
+func TestServiceToServiceMapDualStack(t *testing.T) {
+	svcChangeMap := newServiceChangeMap(false, nil)
+
+	testSvcName := "testDualStack"
+	testSvcNamespace := "test"
+	testSvcPort := int32(12345)
+	testSvcPortName := "testPort"
+	testSvcProtocol := api.ProtocolTCP
+	testSvcNamespacedName := types.NamespacedName{Namespace: testSvcNamespace, Name: testSvcName}
+	testSvcMapKey := proxy.ServicePortName{NamespacedName: testSvcNamespacedName, Port: testSvcPortName}
+
+	testClusterIPv4 := "10.0.0.1"
+	testExternalIPv4 := "8.8.8.8"
+	testSourceRangeIPv4 := "0.0.0.0/1"
+	testClusterIPv6 := "2001:db8:85a3:0:0:8a2e:370:7334"
+	testExternalIPv6 := "2001:db8:85a3:0:0:8a2e:370:7335"
+	testSourceRangeIPv6 := "2001:db8::/32"
+
+	testCases := []struct {
+		desc       string
+		newService *api.Service
+		isIPv6Mode bool
+		// This unit test only does verification on externalIPs and loadBalancerSourceRanges.
```

Comment:
Why the test doesn't cover ClusterIP?

Reply:
Sorry, the comments are confusing, it does cover ClusterIP. Will clarify it a bit in comment,